### PR TITLE
Azure Monitor: Add feature gating for new MetricsQueryEditor with resource picker

### LINKED ISSUE #46058: Create a new Metrics Query that is Feature Gated for Grafana 9

### PR DESCRIPTION
**Pull request by sarahzinger**
**What this PR does / why we need it**:
We are hoping to add the Resource Picker to Metrics Queries for Azure Monitor, but this will be a bigger project over several PRs, with the potential for breaking changes, so we'd like to develop it under a feature gate if possible. This allows us to start that work 

**Which issue(s) this PR fixes**:

Fixes #46058

**Special notes for your reviewer**:

### CODE REVIEW

**Review by sunker — APPROVED** [review]
Nice!

**Review by andresmgot — CHANGES_REQUESTED** [review]
Nice to have this since the beginning! I think we should use [Grafana feature toggles](https://grafana.com/docs/grafana/latest/packages_api/data/featuretoggles/) for this, mostly because this can be available before or after Grafana 9 (and even if it's ready, some people may need to disable it while we fix bugs).

See here some examples: https://github.com/grafana/grafana/blob/main/pkg/services/featuremgmt/registry.go. The `RequiresDevMode` setting seems pretty neat so we can merge temporary code like this one that cannot be used in production.

**Comment by sarahzinger** [comment]
@andresmgot thanks for the suggestion, very cool stuff! This is my first time using these toggles, so let me know if it looks like I'm using them right :)

File: packages/grafana-data/src/types/featureToggles.gen.ts [hunk 1]
```diff
@@ -47,4 +47,5 @@ export interface FeatureToggles {
   annotationComments?: boolean;
   migrationLocking?: boolean;
   fileStoreApi?: boolean;
+  resourcePickerForMetrics?: boolean;
```
Comment (andresmgot) [inline packages/grafana-data/src/types/featureToggles.gen.ts:50]: lol everybody is using a different case here (camelCase, snake_case, kebab-case...). Anyway, I would put "azure" somewhere in the name so it's clear its scope.

File: pkg/services/featuremgmt/registry.go [hunk 2]
```diff
@@ -168,5 +168,10 @@ var (
 			State:           FeatureStateAlpha,
 			RequiresDevMode: true,
 		},
+		{
+			Name:        "resourcePickerForMetrics",
+			Description: "New UI for Azure Monitor Metrics Query",
+			State:       FeatureStateAlpha,
```
Comment (andresmgot) [inline pkg/services/featuremgmt/registry.go:174]: what about setting `RequiresDevMode: true` so we make sure nobody uses it before it's ready? Also we can use `FrontendOnly: true` but I am not really sure what that does.

**Review by andresmgot — APPROVED** [review]
LGTM! Just some minor comments

**Pull request merged**